src/main/java/com/hubspot/maven/plugins/dependency/management/DependencyManagementAnalyzer.java

```diff
@@ -48,24 +64,44 @@ private boolean checkDependencyManagement() {
         if (!projectVersion.equals(managedVersion)) {
           String errorFormat = "Version mismatch for %s, managed version %s does not match project version %s";
           log.warn(String.format(errorFormat, dependencyKey, managedVersion, projectVersion));
+          dependencyVersionMismatchError = true;
```

Comment:
Another option could be to just add to the list here (or maybe after setting success to false) which saves on a bunch of variables:
```java
if (requireManagement.dependencyVersionMismatchMessage() != null) {
  errorMessages.add("Found versions mismatches in managed dependencies:");
  errorMessages.add("  " + requireManagement.dependencyVersionMismatchMessage());
}
```

Up to you.

Reply:
If we add it there, it's getting added multiple times for multiple errors of the same type. So, added it at the end of all checks.

Reply:
Ahh right, good call